src/Workspaces/CSharp/Portable/CodeGeneration/MethodGenerator.cs

```diff
@@ -291,6 +291,14 @@ internal static class MethodGenerator
                 }
             }
 
+            if (destination is CodeGenerationDestination.CompilationUnit)
```

Comment:
Why not replace the following condition (around line 239):

```csharp
                if (destination is not CodeGenerationDestination.CompilationUnit and
                    not CodeGenerationDestination.Namespace and
                    not CodeGenerationDestination.InterfaceType)
```

with

```csharp
                if (destination is not CodeGenerationDestination.Namespace and
                    not CodeGenerationDestination.InterfaceType)
```

Reply:
so compilation unit locations can't support things like accessibility.